make html fails with `scikits.odes` error

@moorepants I'm trying to build the html on my iMac running macOS 10.15.7. It's a late 2013 model so I can't install the latest macOS.

I've consistently been running into a problem with `scikits.odes`. Initially, I created a Python 3.10 venv and pip installed the packages listed in `multibody-book-env.yml` since that's what I normally use for development. That failed with a `scikits.odes` error. I then thought perhaps I needed to roll back to Python 3.9 since that's what's listed in the yml file. That too failed with the same error. Finally I tried installing Anaconda. I installed it using brew since that's how I manage packages on my Mac. I created the conda environment as per the `README.rst` but the `make html` command failed again with an error related to `scikits.odes`. Any idea what's going wrong? Perhaps my macOS level is not supported by `scikits.odes`?

Here's the error messages from `make html`:

```
(multibody-book) riemann:learn-multibody-dynamics arthurryman$ make html
Running Sphinx v5.0.2
making output directory... done
loading intersphinx inventory from https://matplotlib.org/stable/objects.inv...
loading intersphinx inventory from https://numpy.org/doc/stable/objects.inv...
loading intersphinx inventory from https://docs.scipy.org/doc/scipy/objects.inv...
loading intersphinx inventory from https://docs.sympy.org/latest/objects.inv...
loading intersphinx inventory from https://pythreejs.readthedocs.io/en/stable/objects.inv...
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 44 source files that are out of date
updating environment: [new config] 44 added, 0 changed, 0 removed
executing angular. [  4%] angular                                                                                                                                                                                                  
executing configuration%] configuration                                                                                                                                                                                            
executing differentiation differentiation                                                                                                                                                                                          
executing eomes... [ 11%] eom                                                                                                                                                                                                      
executing generalized-forcesneralized-forces                                                                                                                                                                                       
executing holonomic-eom%] holonomic-eom                                                                                                                                                                                            
WARNING: unknown directive or role name: jupyter-download:script
WARNING: unknown directive or role name: jupyter-download:notebook
WARNING: unknown directive or role name: jupyter-download:script
WARNING: unknown directive or role name: jupyter-download:notebook
WARNING: unknown directive or role name: jupyter-download:script
WARNING: unknown directive or role name: jupyter-download:notebook
WARNING: unknown directive or role name: jupyter-download:script
WARNING: unknown directive or role name: jupyter-download:notebook
/Users/arthurryman/Documents/repositories/agryman/learn-multibody-dynamics/generalized-forces.rst:688: WARNING: duplicate label of equation eq-nonholonomic-gaf, other instance in generalized-forces
/Users/arthurryman/Documents/repositories/agryman/learn-multibody-dynamics/generalized-forces.rst:3: CRITICAL: Duplicate ID: "equation-eq-nonholonomic-gaf".
WARNING: unknown directive or role name: jupyter-download:script
WARNING: unknown directive or role name: jupyter-download:notebook

Extension error:
Cell raised uncaught exception:
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
Input In [47], in <cell line: 1>()
----> 1 from scikits.odes import dae

File /usr/local/anaconda3/envs/multibody-book/lib/python3.9/site-packages/scikits/odes/__init__.py:17, in <module>
      1 # -*- coding: utf-8 -*-
      2 """
      3 scikits.odes is a scikit offering a cython wrapper around some extra ode/dae
      4 solvers, so they can mature outside of scipy.
   (...)
     14 .. _SUNDIALS: https://computation.llnl.gov/casc/sundials/main.html
     15 """
---> 17 from .dae import *
     18 from .ode import *
     20 __all__ = ['test'] + [s for s in dir() if not s.startswith('_')]

File /usr/local/anaconda3/envs/multibody-book/lib/python3.9/site-packages/scikits/odes/dae.py:31, in <module>
     28 __version__ = "$Id$"
     29 __docformat__ = "restructuredtext en"
---> 31 from numpy import asarray, array, zeros, sin, int32, isscalar, empty, alen
     32 from copy import copy
     33 import re, sys

ImportError: cannot import name 'alen' from 'numpy' (/usr/local/anaconda3/envs/multibody-book/lib/python3.9/site-packages/numpy/__init__.py)
make: *** [html] Error 2
```

--- a/multibody-book-env.yml
+++ b/multibody-book-env.yml
@@ -6,7 +6,7 @@ dependencies:
   - graphviz
   - jupyter-sphinx
   - matplotlib
-  - numpy
+  - numpy <1.23  # scikits.odes 2.6.3 breaks with numpy 1.23
   - python ==3.9.*
   - pythreejs
   - scikits.odes

